### PR TITLE
[ISSUE #3224]nacos-client module http client replace

### PR DESCRIPTION
**Pull request by Maijh97**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

#3224 

## Brief changelog

Replace the http client in the config package of the `nacos-client` module

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Comment by Maijh97** [comment]
please help to review this pr, thanks @KomachiSion

File: client/src/main/java/com/alibaba/nacos/client/utils/EnvUtil.java [hunk 1]
```diff
@@ -30,45 +28,42 @@
     
     public static final Logger LOGGER = LogUtils.logger(EnvUtil.class);
     
-    public static void setSelfEnv(Map<String, List<String>> headers) {
+    public static void setSelfEnv(Header headers) {
```
Comment (KomachiSion) [inline client/src/main/java/com/alibaba/nacos/client/utils/EnvUtil.java:31]: Is these method refactor correctly? I see the input is come from httpclient.

File: client/src/main/java/com/alibaba/nacos/client/config/http/ServerHttpAgent.java [hunk 2]
```diff
@@ -358,29 +354,37 @@ public void start() throws NacosException {
...
             if (signHeaders != null) {
-                newHeaders.addAll(signHeaders);
+                header.addAll(signHeaders);
             }
         }
-        return newHeaders;
+        String ts = String.valueOf(System.currentTimeMillis());
+        String token = MD5Utils.md5Hex(ts + ParamUtil.getAppKey(), Constants.ENCODE);
+        
+        header.addParam(Constants.CLIENT_APPNAME_HEADER, ParamUtil.getAppName());
```
[the remark this answers is not in the document]
Comment (Maijh97) [reply client/src/main/java/com/alibaba/nacos/client/config/http/ServerHttpAgent.java:377]: This logic comes from the old HttpSimpleClient#setHeaders method

File: client/src/main/java/com/alibaba/nacos/client/utils/EnvUtil.java (hunk 1 above)
Comment (Maijh97) [reply client/src/main/java/com/alibaba/nacos/client/utils/EnvUtil.java:31]: Correct, the original input is also from the http client, this method is called by the ServerListManager#getApacheServerList method

**Pull request merged**